packages/nimble-components/src/table/tests/table.pageobject.ts

```diff
@@ -96,6 +109,95 @@ export class TablePageObject<T extends TableRecord> {
         return rows.item(rowIndex).recordId;
     }
 
+    public getRowWidth(): number {
+        const tableRowContainer = this.tableElement.shadowRoot!.querySelector(
+            '.table-row-container'
+        );
+        return tableRowContainer!.scrollWidth;
+    }
+
+    public getColumnFractionalWidth(columnIndex: number): number {
+        if (columnIndex >= this.tableElement.columns.length) {
+            throw new Error(
+                'Attempting to index past the total number of columns'
+            );
+        }
+
+        return this.tableElement.columns[columnIndex]!.currentFractionalWidth;
+    }
+
+    public getColumnPixelWidth(columnIndex: number): number | undefined {
+        if (columnIndex >= this.tableElement.columns.length) {
+            throw new Error(
+                'Attempting to index past the total number of columns'
+            );
+        }
+
+        return this.tableElement.columns[columnIndex]!.currentPixelWidth;
+    }
+
+    public getCellRenderedWidth(columnIndex: number, rowIndex = 0): number {
+        if (columnIndex >= this.tableElement.columns.length) {
+            throw new Error(
+                'Attempting to index past the total number of columns'
+            );
+        }
+
+        const rows = this.tableElement.shadowRoot!.querySelectorAll('nimble-table-row');
+        if (rowIndex >= rows.length) {
+            throw new Error(
+                'Attempting to index past the total number of rendered rows'
+            );
+        }
+        const row = rows[rowIndex];
+        const cells = row?.shadowRoot?.querySelectorAll('nimble-table-cell');
+        if (columnIndex >= (cells?.length ?? 0)) {
+            throw new Error(
+                'Attempting to index past the total number of cells'
+            );
+        }
+
+        const columnCell = cells![columnIndex]!;
+        return columnCell.getBoundingClientRect().width;
+    }
+
+    public setColumnFractionalWidth(
+        columnIndex: number,
+        fractionalWidth: number
+    ): void {
+        if (columnIndex >= this.tableElement.columns.length) {
+            throw new Error(
+                'Attempting to index past the total number of columns'
+            );
+        }
+
+        this.tableElement.columns[columnIndex]!.internalPixelWidth = undefined;
```

Comment:
Question coming from a place of ignorance: why is the page object using "internal" APIs to set widths? Could it use the public APIs instead? If not, how would we expect clients to set these values (from production TS code, where they couldn't use our page object)?

Reply:
So, the public APIs are non-existent on the base `TableColumn`. They are only surfaced as public APIs on the concrete column types (like `TableColumnText`). So, you bring up an interesting point, in that the pageObject, as implemented, might suggest that any column would support fractional widths, which isn't the case.

The reasoning behind this approach, from a test perspective at least, not necessarily the pageObject API, was to handle all of the various state scenarios that the `TableColumn` could theoretically be configured to support. I think what the correct thing to do, is to remove this from the pageObject, and just do this action within the test itself.

Reply:
Ultimately, I think the right approach will be to have a page object for each component type, and not have text-column-specific methods on the `TablePageObject` as that will potentially blow up its API once we introduce other column types. For now, in order to unblock other efforts depending on this PR, I think it prudent to create a Tech Debt task to create the `TableColumnText` page object class.